src/modules/error.js

```diff
@@ -53,6 +57,46 @@ export class VisualizationNotFoundError extends VisualizationError {
     }
 }
 
+export class NoProgramError extends VisualizationError {
+    constructor() {
+        super(
+            EmptyBox,
+            i18n.t('No program selected'),
+            i18n.t('Choose a program from the Program Dimensions sidebar.')
+        )
+    }
+}
+
+export class NoColumnsError extends VisualizationError {
+    constructor() {
+        super(
+            EmptyBox,
+            i18n.t('Columns is empty'),
+            i18n.t('Add at least one item to Columns.')
+        )
+    }
+}
+
+export class NoOrgUnitError extends VisualizationError {
+    constructor() {
+        super(
+            DataError,
+            i18n.t('No organisation unit selected'),
+            i18n.t('Add at least one organisation unit to the layout.')
+        )
+    }
+}
+
+export class NoPeriodError extends VisualizationError {
+    constructor() {
+        super(
+            PeriodError,
+            i18n.t('No time dimension selected'),
+            i18n.t('Add at least one time dimension to the layout.')
+        )
+    }
+}
+
```

Comment:
Could be there are reasons for going with this class based approach I'm not aware of, but I find this rather confusing. Because these are classes with an `Error` suffix, we see them invoked as `new BlaBlaError()`. This firstly makes me think that these classes are [custom error classes](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/error#custom_error_types) which we could throw, and secondly makes me suspect the returned object comes with methods we might want to call.

It seems to me that all we want to have is a bunch of simple functions that produce simple objects with properties `icon`, `title`, and `description`. If this is in fact all we need, I would be in favour of abandoning this class based approach in favour of some simple functions that return object.

Reply:
I think the class based approach (originally in DV https://github.com/dhis2/data-visualizer-app/blob/master/packages/app/src/modules/error.js) was to unify the errors, both from an input (`visType`) and output (params) perspective. But as we don't use any input in ER (as all errors use static text) the only thing it really does is to make sure what to expect back from the object (`title`, `icon`, `description`). Im not sure if class is the best way to do this though, but having some sort of typed interface would be convenient I suppose so we have a clear pattern to follow. Suggestions are welcome

Reply:
We adopted many things from DV, but we don't have to keep them if there are simpler and better solutions.
I always thought ER as a opportunity to review stuff we did in DV and improve and eventually bring those improvements back to DV.

Reply:
I'm not sure what you mean about "typing", but I'll just give a very brief demo of what I was trying to suggest:
```js
// CURRENT
export class VisualizationError {
    constructor(icon, title, description) {
        this.icon = icon
        this.title = title
        this.description = description
    }
}

export class GenericClientError extends VisualizationError {
    constructor() {
        super(
            GenericError,
            genericErrorTitle,
            i18n.t('There is a problem with this visualization.')
        )
    }
}
// SUGGESTION
export const visualizationError (icon, title, description) => ({
    icon,
    title,
    description,
})

export const genericClientError () => visualizationError({
    GenericError,
    genericErrorTitle,
    i18n.t('There is a problem with this visualization.')
})
```

Reply:
Fixed with https://github.com/dhis2/event-reports-app/pull/934/commits/db6096acf1c240e3b0a56494f4633d33cf6f1881